src/components/slide-toggle/slide-toggle.ts

```diff
@@ -84,6 +92,23 @@ export class MdSlideToggle implements AfterContentInit, ControlValueAccessor {
     this._isInitialized = true;
   }
 
+  /** TODO: internal */
+  ngAfterViewInit() {
+    let hammerManager = new Hammer(this._thumbContainerRef.nativeElement);
+
+    // The Pan / Drag recognition should immediately detect the starting drag event, without
+    // waiting for the drag to exceed the elements width.
+    hammerManager.add(new Hammer.Pan({
+      threshold: 0,
+      pointers: 0,
+      direction: Hammer.DIRECTION_HORIZONTAL
+    }));
+
+    hammerManager.on('panstart', this._onDragStart.bind(this));
+    hammerManager.on('panmove', this._onDrag.bind(this));
+    hammerManager.on('panend', this._onDragEnd.bind(this));

```

Comment:
> Why are you adding the events directly instead of using handlers defined via the template, e.g.,
Yeah great idea - as we discussed on Slack

> Also, I tried this on my phone (Galaxy S6) and it was a bit sluggish. I wonder if we should throttle these drag events (would we need a throttle function in the library, though...)
Yeah I fixed that by calculating the absolute difference (as done in the slider)

> Actually, it's fine for now- can you just add a TODO to throttle the events later?
Not sure, why throttling is necessary here? It seems like throttling would probably make it less smooth.